crates/chain/src/chain_data.rs

```diff
@@ -15,25 +15,45 @@ use crate::{Anchor, COINBASE_MATURITY};
     ))
 )]
 pub enum ChainPosition<A> {
-    /// The chain data is seen as confirmed, and in anchored by `A`.
-    Confirmed(A),
-    /// The chain data is not confirmed and last seen in the mempool at this timestamp.
-    Unconfirmed(u64),
+    /// The chain data is confirmed as it is anchored in the best chain by `A`.
+    Confirmed {
+        /// The [`Anchor`].
+        anchor: A,
+        /// Whether the chain data is anchored transitively by a child transaction.
+        ///
+        /// If the value is `Some`, it means we have incomplete data. We can only deduce that the
+        /// chain data is confirmed at a block equal to or lower than the block referenced by `A`.
+        transitively: Option<Txid>,
+    },
+    /// The chain data is not confirmed.
+    Unconfirmed {
+        /// When the chain data is last seen in the mempool (if seen at all).
+        ///
+        /// It may be possible that the chain data was last seen as confirmed in a conflicting
+        /// chain.
```

Comment:
Would this more clearly describe when this value is Some or None? 
```suggestion
        /// If the value is `Some` then this is when the chain data was last seen in a mempool. 
        /// If `None` then it was only seen as confirmed in a conflicting chain.
```

Reply:
```
        /// When the chain data is last seen in the mempool.
        ///
        /// This value will be `None` if the chain data was never seen in the mempool and only seen
        /// in a conflicting chain.
```

How about this?

Reply:
Yes this is clear and more concise.